data/amm.go

```diff
@@ -0,0 +1,32 @@
+package data
+
+import "fmt"
+
+var txAmmAcceptedMap = map[TransactionType]bool{AMM_DEPOSIT: true, AMM_WITHDRAW: true, AMM_CREATE: true, AMM_VOTE: true, AMM_BID: true, PAYMENT: true}
+
+func (txm *TransactionWithMetaData) AMM() (*AMM, error) {
```

Comment:
What does this function do? Return the first found AffectedNode which is an AMM struct? Is there a situation where there might be two affected AMM structs, perhaps in a Payment transaction result?